# Lab Instructions

Find a dataset that interests you. I'd recommend starting on [Kaggle](https://www.kaggle.com/). Read through all of the material about the dataset and download a .CSV file.

1. Write a short summary of the data.  Where did it come from?  How was it collected?  What are the features in the data?  Why is this dataset interesting to you?  

2. Identify 5 interesting questions about your data that you can answer using Pandas methods.  

3. Answer those questions!  You may use any method you want (including LLMs) to help you write your code; however, you should use Pandas to find the answers.  LLMs will not always write code in this way without specific instruction.  

4. Write the answer to your question in a text box underneath the code you used to calculate the answer.



1. The dataset used in this analysis is the AI Worker Burnout and Attrition Risk Dataset from Kaggle.
Dataset link: https://www.kaggle.com/datasets/nudratabbas/ai-worker-burnout-and-attrition-risk-dataset

This dataset contains 1,500 records of workers in AI-related roles and includes information about their work environment, AI usage, productivity, burnout levels, and likelihood of leaving their job.

The data was collected and compiled to analyze how AI adoption affects employee productivity, burnout, and job satisfaction. It appears to simulate or aggregate workplace analytics data across multiple industries and countries.

The dataset includes 21 features of data.  

This dataset is interesting because it explores the relationship between artificial intelligence adoption and employee wellbeing. As AI becomes more integrated into workplaces, it is important to understand whether it improves productivity, increases burnout, or influences employee turnover.

2.
    1. Which job roles have the highest burnout scores?
    2. Does using AI for more hours per day improve productivity?
    3. What is the distribution of attrition risk among employees?
    4. Is there a relationship between burnout and job satisfaction?
    5. How does salary differ by education level?

Question 1: Which job roles have the highest burnout scores? 

In [7]:
import pandas as pd

df = pd.read_csv("ai_worker_burnout_attrition_2026.csv")

df.groupby("job_role")["burnout_score"].mean().sort_values(ascending=False).head(5)

job_role
Software Engineer    51.266667
DevOps Engineer      50.931034
Data Analyst         50.926230
Backend Engineer     50.921260
ML Engineer          50.697248
Name: burnout_score, dtype: float64

Answer 1: Software Engineer, DevOps Engineer, Data Analyst, Backend Engineer, ML Engineer

Question 2: Does using AI for more hours per day improve productivity?

In [15]:
bins = [0,1,2,3,4,5,6,10]
labels = ["0-1","1-2","2-3","3-4","4-5","5-6","6+"]

df["ai_hours_bucket"] = pd.cut(
    df["hours_with_ai_assistance_daily"],
    bins=bins,
    labels=labels,
    include_lowest=True
)

df.groupby("ai_hours_bucket", observed=False)["productivity_score"].mean()

ai_hours_bucket
0-1    56.818182
1-2    57.497462
2-3    57.192893
3-4    56.594059
4-5    57.470320
5-6    57.692708
6+     58.422977
Name: productivity_score, dtype: float64

Answer 2: Employees who spend more hours working with AI tend to have slightly higher productivity scores. Workers using AI for 6+ hours daily had the highest productivity (~58.4), while those using it less frequently averaged slightly lower scores.

This suggests that AI assistance may help workers complete tasks more efficiently.

Question 3: What is the distribution of attrition risk among employees?

In [12]:
df["attrition_risk"].value_counts()

attrition_risk
Low       724
Medium    691
High       85
Name: count, dtype: int64

Answer 3: The dataset shows the following distribution:
Low risk: 724 employees
Medium risk: 691 employees
High risk: 85 employees
Most employees fall into the low or medium attrition risk categories, meaning the majority are relatively stable in their jobs.

Question 4: Is there a relationship between burnout and job satisfaction?

In [13]:
df[["burnout_score", "job_satisfaction_1_5"]].corr()

,burnout_score,job_satisfaction_1_5
burnout_score,1.000000,-0.629328
job_satisfaction_1_5,-0.629328,1.000000


Answer 4: The correlation between burnout and job satisfaction is approximately -0.63.

This indicates a strong negative relationship, meaning:

Higher burnout -> Lower job satisfaction

Lower burnout -> Higher job satisfaction

This aligns with typical workplace research findings.

Question 5: How does salary differ by education level?

In [14]:
df.groupby("education_level")["salary_usd_k"].mean().sort_values(ascending=False)

education_level
Bootcamp       145.091549
Master         141.427245
PhD            140.459184
Bachelor       139.203822
Self-taught    137.575439
Name: salary_usd_k, dtype: float64

Answer 5: Average salary by education level:
Bootcamp: ~$145k
Master’s: ~$141k
PhD: ~$140k
Bachelor’s: ~$139k
Self-taught: ~$137k

Interestingly, bootcamp graduates had the highest average salary in this dataset, which may reflect the strong demand for specialized AI and technical skills.